nixos/modules/services/backup/restic.nix

```diff
@@ -121,6 +149,29 @@ with lib;
           extraOptions = concatMapStrings (arg: " -o ${arg}") backup.extraOptions;
           connectTo = elemAt (splitString ":" backup.repository) 1;
           resticCmd = "${pkgs.restic}/bin/restic${extraOptions}";
+          filesFromTmpFile = "/tmp/restic-backup-${name}-files";
+          preStartInit = if backup.initialize
+                         then "${resticCmd} snapshots || ${resticCmd} init"
+                         else "";
+          preStartFiles = if backup.dynamicFilesFromCmd != null
+                          then "${backup.dynamicFilesFromCmd} > ${filesFromTmpFile}"
```

Comment:
It might make sense to put the command in a separate script and pipe the script output to a temporary file.

Reply:
Then also more complex expressions that spawn across multiple lines work:

```
find /home/matt/git -type d -name .git
find /home/matt/homework -type d -name .zip
```